core/templates/components/main-nav/main-nav.twig

```diff
@@ -0,0 +1,74 @@
++ *
++ */
++#}
+<nav {{- attributes }} class="su-main-nav no-js {{ modifier_class }}" aria-label="{{ aria_label|default('main menu') }}">
+  <button class="su-main-nav__toggle {{- toggle_modifier_class }}" aria-expanded="false">{{ toggle_text }}</button>
+  <ul class="su-main-nav__menu-lv1">
+    {% spaceless %}
+    {% if list_items is iterable %}
+      {# If no custom markup, generate markup based on structured data #}
+      {% for item in list_items %}
+        {% set is_parent = item.lv2 is empty ? false : true %}
+        <li class="{% if is_parent %}su-main-nav__item--parent{% endif %} {% if item.current %}su-main-nav__item--current{% endif %}">
+          {% set the_attrs = "" %}
+          {% if item.attributes is not empty and item.attributes is iterable %}
+            {% for pair in item.attributes %}
+              {% set the_attrs = the_attrs ~ ' ' ~ pair.attr ~ '="' ~ pair.value ~ '"' %}
+            {% endfor %}
+          {% endif %}
+          <a href="{{ item.href }}" {{ the_attrs }}>{{ item.text }}</a>
+          {% if is_parent -%}
+            <ul class="su-main-nav__menu-lv2">
+              {%- for item_lv2 in item.lv2 -%}
+                <li class="{% if item_lv2.current %}su-main-nav__item--current{% endif %}">
+                  {% set the_attrs = "" %}
+                  {% if item_lv2.attributes is not empty and item_lv2.attributes is iterable %}
+                    {% for pair in item_lv2.attributes %}
+                      {% set the_attrs = the_attrs ~ ' ' ~ pair.attr ~ '="' ~ pair.value ~ '"' %}
+                    {% endfor %}
+                  {% endif %}
+                  <a href="{{ item_lv2.href }}" {{ the_attrs }}>{{ item_lv2.text }}</a>
+                </li>
+              {% endfor %}
+            </ul>
+          {% endif -%}
+        </li>
+      {% endfor %}
+    {% else %}
+      {# If custom markup is provided, emit it as is #}
+      {{ list_items }}
+    {% endif %}
+    {% endspaceless %}
+  </ul>
+  {% if "su-main-nav--mobile-search" in modifier_class %}
+    {# no blessed idea why {% include ... with mobile_search %} doesn't work, but it doesn't, so reference each variable individually :( #}
+    {% include "@decanter/components/site-search/site-search.twig" with
+      {
+        "action": mobile_search.action,
+        "method": mobile_search.method,
+        "search_label": mobile_search.search_label,
+        "placeholder": mobile_search.placeholder,
+        "search_input_name": mobile_search.search_input_name,
+        "search_icon_src": mobile_search.search_icon_src
+      }
+      only
+    %}
+  {% endif %}
+</nav>
+<script>
+  // Do this immediately to minimize FOUC.
+  NodeList.prototype.forEach = NodeList.prototype.forEach || Array.prototype.forEach;
+  document.querySelectorAll('nav.su-main-nav').forEach(nav => { nav.classList.remove('no-js'); } );
```

Comment:
This gets fired off 5 times in the styleguide. Each time the nav is rendered this code gets introduced. This also collectively picks up a nav each time it runs. I suppose that isn't harmful but does have me itching for an optimization. Good for now.